### PR TITLE
feat(selection): Remove nodes based on tag name and class.

### PR DESCRIPTION
**Pull request by jenstroeger**
Extend the [selection.removeFormatting()](https://github.com/livingdocsIO/editable.js/blob/77e45729651eda118ec232c0adf603171d7686b0/src/selection.js#L202-L206) function so that it can remove nodes based on tag name _and_ class name; handy if I’d like to remove e.g. `span.foo` only.

### CODE REVIEW

File: src/content.js [hunk 1]
```diff
@@ -310,18 +310,19 @@ export function removeFormattingElem (host, range, elem) {
...
 
 // Unwrap all tags this range is affected by.
 // Can also affect content outside of the range.
-export function nuke (host, range, tagName) {
+export function nuke (host, range, tagName, className) {
   getTags(host, range).forEach((elem) => {
     if (elem.nodeName.toUpperCase() !== 'BR' &&
-      (!tagName || elem.nodeName.toUpperCase() === tagName.toUpperCase())) {
+      (!tagName || (elem.nodeName.toUpperCase() === tagName.toUpperCase() &&
+        (!className || elem.classList.contains(className))))) {
```
Comment (marcbachmann) [inline src/content.js:325]: We could use a selector here instead of a fixed class name. That way you can pass `.someclass` and also attributes `[data-someattribute]`.

Comment (jenstroeger) [reply src/content.js]: Great suggestion, thanks @marcbachmann! Commit https://github.com/livingdocsIO/editable.js/pull/228/commits/1f008690ce35f9304d63910a4577a8cc7cb14f1c.

**Review by peyerluk — APPROVED** [review]

File: src/content.js [hunk 2]
```diff
@@ -310,18 +310,19 @@ export function removeFormattingElem (host, range, elem) {
   })
 }
 
-export function removeFormatting (host, range, tagName) {
+export function removeFormatting (host, range, tagName, selector) {
```
Comment (peyerluk) [inline src/content.js:313]: We could shorten this to: `export function removeFormatting (host, range, selector) {`

This would still be compatible with the old version.

Comment (jenstroeger) [reply src/content.js]: Good point, thanks @peyerluk! Commit https://github.com/livingdocsIO/editable.js/pull/228/commits/56491acacdaecd2537eea8197ccafd809ca7f35a.

File: src/content.js [hunk 3]
```diff
@@ -310,18 +310,17 @@ export function removeFormattingElem (host, range, elem) {
...
 }
 
 // Unwrap all tags this range is affected by.
 // Can also affect content outside of the range.
-export function nuke (host, range, tagName) {
+export function nuke (host, range, selector) {
   getTags(host, range).forEach((elem) => {
-    if (elem.nodeName.toUpperCase() !== 'BR' &&
-      (!tagName || elem.nodeName.toUpperCase() === tagName.toUpperCase())) {
+    if (elem.nodeName.toUpperCase() !== 'BR' && (!selector || elem.matches(selector))) {
```
Comment (jenstroeger) [inline src/content.js:323]: An alternative implementation of that function would be:
```javascript
export function nuke (host, range, selector) {
  if (!selector) selector = "br"
  getTags(host, range).forEach((elem) => {
    if (elem.matches(selector)) unwrap(elem)
  })
}
```

**Review by marcbachmann — APPROVED** [review]
